apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.html.ts

```diff
@@ -41,106 +42,135 @@ export function AiGeneratePage({ resLocals }: { resLocals: Record<string, any> }
-                  </label>
-                  <textarea name="prompt_grading" id="user-prompt-llm-grading" class="form-control">
-                  </textarea>
-                  <div class="form-text form-muted">
-                    <em>
-                      Example: the answer is computed as sqrt(2 * h / g) where g = 9.81 m/s^2
-                    </em>
-                  </div>
-                </div>
-                <button class="btn btn-primary">
-                  <span
-                    class="spinner-grow spinner-grow-sm d-none"
-                    role="status"
-                    aria-hidden="true"
-                    data-loading-class-remove="d-none"
-                  >
-                  </span>
-                  Create question
-                </button>
-              </form>
-              Or choose a question from our list of example prompts:
-              <select
-                id="user-prompt-example"
-                onchange="setPromptToExample()"
-                class="custom-select"
-              >
-                <option value=""></option>
-                ${examplePrompts.map(
-                  (question) =>
-                    html`<option
-                      value="${question.id}"
-                      data-prompt-general="${question.promptGeneral}"
-                      data-prompt-user-input="${question.promptUserInput}"
-                      data-prompt-grading="${question.promptGrading}"
-                    >
-                      ${question.id}
-                    </option>`,
-                )}
-              </select>
-              <div id="generation-results"></div>
-              <br />
-              <div>
-                <a href="${resLocals.urlPrefix}/ai_generate_question_jobs" class="btn btn-primary">
-                  Previous Generation Jobs
-                </a>
-              </div>
+        <main id="content" class="container mb-4">
+          <div class="mb-3">
+            <a
+              href="${resLocals.urlPrefix}/ai_generate_question_drafts"
+              class="btn btn-sm btn-primary"
+            >
+              <i class="fa fa-arrow-left" aria-hidden="true"></i>
+              Back to draft questions
+            </a>
+          </div>
+          <div class="card">
+            <div
+              class="card-header bg-primary text-white d-flex align-items-center justify-content-between"
+            >
+              Generate question with AI
             </div>
+            <div class="card-body">
+                    <div id="generation-prompter">
+                      <p>
+                        Describe your question in as much detail as possible in the text boxes
+                        below.
+                      </p>
+                      <form
+                        name="add-question-form"
+                        hx-post="${resLocals.urlPrefix}/ai_generate_question"
```

Comment:
Is using HTMX really necessary here? It seems like we're just relying on a redirection after the response, which we could do without HTMX here.

Reply:
Answered my own question: yes, we use it to show errors, I think.